apps/web/Dockerfile

```diff
@@ -1,24 +1,55 @@
-FROM node:20-alpine AS build
+FROM node:20-alpine AS base
 
+
+FROM base AS builder
+RUN apk add --no-cache libc6-compat
+RUN apk update
+# working directory
 WORKDIR /app
+RUN npm install -g turbo
+COPY . .
+RUN turbo prune web --docker
 
-RUN npm i -g pnpm
+# Add lockfile and package.json's of isolated subworkspace
+FROM base AS installer
+RUN apk add --no-cache libc6-compat
+RUN apk update
+WORKDIR /app
 
-# COPY package.json .
-COPY turbo.json .
-COPY pnpm-workspace.yaml .
-COPY package*.json .
-COPY ./packages ./packages
+# install the dependencies
+RUN npm install -g pnpm
+COPY .gitignore .gitignore
+COPY --from=builder /app/out/json/ .
+COPY --from=builder /app/out/pnpm-lock.yaml ./pnpm-lock.yaml
+RUN pnpm  install
 
-COPY apps/web/package.json apps/web/package.json
-COPY apps/web/tsconfig.json apps/web/tsconfig.json
+# Build the project
+COPY --from=builder /app/out/full/ .
+COPY turbo.json turbo.json
 
-RUN pnpm install
 
-COPY . .
+RUN pnpm turbo build --filter=web...
 
-RUN pnpm run build:web
+FROM base AS runner
+WORKDIR /app
 
+# Don't run production as root
+RUN addgroup --system --gid 1001 nodejs
+RUN adduser --system --uid 1001 nextjs
+USER nextjs
+
+COPY --from=installer /app/apps/web/next.config.js .
+COPY --from=installer /app/apps/web/package.json .
+
+# Automatically leverage output traces to reduce image size
+# https://nextjs.org/docs/advanced-features/output-file-tracing
+COPY --from=installer --chown=nextjs:nodejs --chmod=644 /app/apps/web/.next/standalone ./
```

Comment:
sonarqube complains about not giving write permissions. can you confirm on that once?

Reply:
it's suggesting to run as root but according to turbo docs we should not not run production as root